server/src/test/java/org/opensearch/index/shard/SegmentReplicationIndexShardTests.java

```diff
@@ -385,6 +385,71 @@ public void testReplicaRestarts() throws Exception {
         }
     }
 
+    public void testNRTReplicaWithRemoteStorePromotedAsPrimary() throws Exception {
```

Comment:
curious if this test is using local fs repository for uploading / downloading segments?

Reply:
It uses `FsBlobStore`. Please refer `IndexShardTestCase.newRemoteDirectory()` for more details.